libevmasm/Assembly.cpp

```diff
@@ -259,104 +237,63 @@ Json::Value Assembly::assemblyJSON(map<string, unsigned> const& _sourceIndices)
 				sourceIndex = static_cast<int>(iter->second);
 		}
 
-		switch (i.type())
+		auto [name, data] = i.nameAndData();
+		Json::Value item;
+		item["name"] = name;
+		item["begin"] = i.location().start;
+		item["end"] = i.location().end;
+		if (i.m_modifierDepth != 0)
+			item["modifierDepth"] = static_cast<int>(i.m_modifierDepth);
+		std::string jumpType = i.getJumpTypeAsString();
+		if (!jumpType.empty())
+			item["jumpType"] = jumpType;
+		if (name == "PUSHLIB")
+			data = m_libraries.at(h256(data));
+		else if (name == "PUSHIMMUTABLE" || name == "ASSIGNIMMUTABLE")
+			data = m_immutables.at(h256(data));
+		if (!data.empty())
+			item["value"] = data;
+		item["source"] = sourceIndex;
+		code.append(item);
+
+		if (name == "tag")
```

Comment:
I would feel better if this as a condition on the type instead of the returned string. It is much clearer that there can be no collisions.